protocol/avdl/keybase1/metadata.avdl

```diff
@@ -41,10 +41,15 @@ protocol metadata {
          KeyBundle ReaderBundle;
   }
 
+  record LockContext {
+    long    requireLockID;
```

Comment:
It might be nice to typedef a real `MetadataLockID` type instead of using a raw `long`, especially so that the `lockBeforeGet` parameter is more clear.

Why is `long` the right type though?  I was expecting it to be a `bytes` that is some irreversible function of the repo and branch name.  Are you sure that can always map to a `long`?  Either way, there should probably be a comment explaining that it needs to be unique per TLF, perhaps with an example of how to generate one in the case of a git repo.